sbws/lib/v3bwfile.py

```diff
@@ -544,29 +550,56 @@ def from_results(cls, results, secs_recent=None, secs_away=None,
         # In #28563 will include again these lines, but make them unparseable
         # by Tor.
         # This might confirm #28042.
+        exclusion_reason = None
         kwargs['relay_recent_measurement_exclusion_not_success_count'] = \
             len(results) - len(success_results)
         if not success_results:
-            return None, 'recent_measurement_exclusion_not_success_count'
+            exclusion_reason = 'recent_measurement_exclusion_not_success_count'
```

Comment:
I'm concerned by the change in logic by moving all of these "return" statements.  The way that the code was written before, it would return the _first_ reason that it found for excluding the relay.  But now, it looks like it will return the _last_ reason.  Do we really want to make that change here?

Reply:
You are totally right, i left it to return on each condition as it was, just return also the reason